clang/lib/Driver/Driver.cpp

```diff
@@ -1230,7 +1230,8 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
   bool IsSYCL = C.getInputArgs().hasFlag(options::OPT_fsycl,
                                          options::OPT_fno_sycl, false) ||
                 C.getInputArgs().hasArgNoClaim(options::OPT_fsycl_device_only,
-                                               options::OPT_fsyclbin);
+                                               options::OPT_fsyclbin,
+                                               options::OPT_fsyclbin_EQ);
```

Comment:
You can just check for `OPT_fsyclbin_EQ` for this check and all others since `OPT_fsyclbin` is just an alias.